### PR TITLE
use System.Memory on legacy runtimes

### PR DESCRIPTION
**Pull request by SimonCropp**
the plan is this will allow us to use span APIs in more places without conditional compilation

### CODE REVIEW

**Comment by SimonCropp** [comment]
ping @sungam3r

File: src/Serilog/Core/LevelOverrideMap.cs [hunk 1]
```diff
@@ -58,18 +58,15 @@ public LevelOverride(string context, LoggingLevelSwitch levelSwitch)
     }
 
     public void GetEffectiveLevel(
-#if FEATURE_SPAN
         ReadOnlySpan<char> context,
```
Comment (sungam3r) [inline src/Serilog/Core/LevelOverrideMap.cs:61]: I do not understand why span is used here. At the first glance it can be safely changed to string.

**Comment by sungam3r** [comment]
I'm fine with new `System.Memory` conditional dependency. We used the same approach in GraphQL.NET.

**Comment by sungam3r** [comment]
Not applicable?

**Pull request closed without merging**